provisioning/baremetal_pod.go

```diff
@@ -257,7 +268,27 @@ func newMetal3InitContainers(info *ProvisioningInfo) []corev1.Container {
 		initContainers = append(initContainers, createInitContainerStaticIpSet(info.Images, &info.ProvConfig.Spec))
 	}
 
-	initContainers = append(initContainers, createInitContainerMachineOsDownloader(info, true))
+	// If the PreProvisioningOSDownloadURLs are set, we fetch the URLs of either CoreOS ISO and IPA assets or in some
+	// cases only the IPA assets
+	liveURLs := getPreProvisioningOSDownloadURLs(&info.ProvConfig.Spec)
+	if liveURLs != nil {
```

Comment:
It's always non-nil now (which is good!), so this should be `if len(liveURLSs) > 0`

Reply:
Changed.